compiler/src/iree/compiler/Codegen/Common/MaterializeEncodingPass.cpp

```diff
@@ -0,0 +1,344 @@
+    OpBuilder &builder, Location loc,
+    IREE::LinalgExt::MaterializeEncodingTypeConverter &typeConverter,
+    IREE::Flow::DispatchTensorType dispatchTensorType, ValueRange dynamicDims) {
+  auto boundTensorType =
+      dispatchTensorType.getBoundType().dyn_cast<RankedTensorType>();
+  if (!boundTensorType) {
+    return failure();
+  }
+
+  auto encoding =
+      boundTensorType.getEncoding().dyn_cast<IREE::LinalgExt::EncodingAttr>();
+  if (!encoding) {
+    return failure();
+  }
+
+  IREE::LinalgExt::MaterializeEncodingFn materializeEncodingFn =
+      typeConverter.getMaterializeEncodingFn();
+  FailureOr<IREE::LinalgExt::MaterializeEncodingInfo> encodingInfo =
+      materializeEncodingFn(encoding.getEncoding().getValue());
+  if (failed(encodingInfo)) {
+    return failure();
+  }
+
+  SmallVector<OpFoldResult, 4> targetShape = getMixedSizes(
+      builder.getIndexArrayAttr(dispatchTensorType.getShape()), dynamicDims);
+  SmallVector<OpFoldResult> innerTileSizes = llvm::to_vector(llvm::map_range(
+      encodingInfo->innerTileSizes,
+      [&](int64_t v) -> OpFoldResult { return builder.getIndexAttr(v); }));
+  SmallVector<OpFoldResult> convertedTargetShape =
+      IREE::LinalgExt::PackOp::getResultShape(
+          builder, loc, targetShape, innerTileSizes, encodingInfo->innerDimsPos,
+          encodingInfo->outerDimsPerm);
+  return convertedTargetShape;
+}
+
+/// For `dispatchTensorType` that bind a `RankedTensorType` with encoding,
+/// returns the dynamic dimensions of the materialized shape of the
+/// `dispatchTensorType`. The dynamic dimensions of the `dispatchTensorType` are
+/// provided in `dynamicDims`.
+static FailureOr<SmallVector<Value>> getPackedDynamicDimsForDispatchTensor(
+    OpBuilder &builder, Location loc,
+    IREE::LinalgExt::MaterializeEncodingTypeConverter &typeConverter,
+    IREE::Flow::DispatchTensorType dispatchTensorType, ValueRange dynamicDims) {
+  FailureOr<SmallVector<OpFoldResult>> convertedTargetShape =
+      getPackedDimsForDispatchTensor(builder, loc, typeConverter,
+                                     dispatchTensorType, dynamicDims);
+  if (failed(convertedTargetShape)) {
+    return failure();
+  }
+  SmallVector<int64_t> convertedStaticTargetShape;
+  SmallVector<Value> convertedDynamicTargetShape;
+  dispatchIndexOpFoldResults(
+      convertedTargetShape.value(), convertedDynamicTargetShape,
+      convertedStaticTargetShape, ShapedType::kDynamicSize);
+  return convertedDynamicTargetShape;
+}
+
+namespace {
+/// Given the `encoding` return the `MaterializeEncodingInfo` to use for
+/// materializing the pack op.
+// TODO(ravishankarm): THis is currently hard-coded here for convenience. When
+// used in IREE, this will be computed based on the architecture information in
+// `hal.executable.variant`.
+static FailureOr<IREE::LinalgExt::MaterializeEncodingInfo>
+getPackOpInfoFromEncoding(IREE::LinalgExt::TensorEncoding encoding) {
```

Comment:
@bjacob please take a look at this PR and review it. Cause this is probably is pretty close to the place you have been working. Also this specific method is ad-hoc. I am hoping that this is something that you can help develop further that returns the actual values based on `hal.executable.variant`.

Reply:
I'll add a little helper that makes finding the configuration easier. I've got a trick I've been using in stream/hal for doing that which allows for annotations in tests that don't require adding all the boilerplate ops, e.g.:
https://github.com/iree-org/iree/blob/b28b2c22ee8342a63b0b73f2befc4b101dc09612/compiler/src/iree/compiler/Dialect/Stream/Transforms/test/pack_constants.mlir#L83

Reply:
https://github.com/iree-org/iree/pull/11057